cloudferrylib/os/network/neutron.py

```diff
@@ -1196,3 +1197,67 @@ def get_new_extnet_id(self, src_net_id, src_nets, dst_nets):
             net_hash = self.get_res_hash_by_id(src_nets, src_net_id)
             dst_net_id = self.get_res_by_hash(dst_nets, net_hash)['id']
         return dst_net_id
+
+    @staticmethod
+    def get_free_ip(subnet, excluded_ips):
+        for pool in subnet['allocation_pools']:
+            addr = ipaddr.IPAddress(pool['start'])
+            end = ipaddr.IPAddress(pool['end'])
+            while addr <= end:
+                if str(addr) not in excluded_ips:
+                    return str(addr)
+                addr += 1
+        raise Exception("Can't find free ip for router")
```

Comment:
Raising `Exception` is considered bad practice. Please define more specific exception type for this case.